src/parser/mod.rs

```diff
@@ -6922,18 +6924,23 @@ impl<'a> Parser<'a> {
         })
     }
 
-    pub fn parse_optional_inline_comment(&mut self) -> Result<Option<CommentDef>, ParserError> {
+    pub fn parse_optional_inline_comment(
+        &mut self,
+        support_dollar_quoted_comment: bool,
+    ) -> Result<Option<CommentDef>, ParserError> {
         let comment = if self.parse_keyword(Keyword::COMMENT) {
             let has_eq = self.consume_token(&Token::Eq);
             let next_token = self.next_token();
-            match next_token.token {
-                Token::SingleQuotedString(str) => Some(if has_eq {
-                    CommentDef::WithEq(str)
-                } else {
-                    CommentDef::WithoutEq(str)
-                }),
+            let comment = match next_token.token {
+                Token::SingleQuotedString(str) => str,
+                Token::DollarQuotedString(str) if support_dollar_quoted_comment => str.value,
```

Comment:
I think no need to guard this with a flag, we can always accept a dollar quoted string if one shows up. Also noticed we have similar [function here](https://github.com/apache/datafusion-sqlparser-rs/blob/24a2968b33787626fe63fc626c19dbfb50ec48f5/src/parser/mod.rs#L5323-L5328) could we maybe pull that part out into a helper and reuse in both places?

Reply:
A parameter `support_dollar_quoted_comment` is removed.